lib/wikidata_ids_decorator.rb

```diff
@@ -1,5 +1,25 @@
 require 'wikidata_ids_decorator/version'
+require 'scraped'
+require 'wikidata/fetcher'
 
 module WikidataIdsDecorator
-  # Your code goes here...
+  class Links < Scraped::Response::Decorator
+    def body
+      Nokogiri::HTML(super).tap do |doc|
+        links = doc.css('#bodyContent a[href*="/wiki"][title]').reject { |a| a[:title].include? ':' }
+        wdids = wikidata_ids(links.map { |a| a[:title] }.uniq)
+        links.each { |a| a[:wikidata] = wdids[a[:title]] }
+      end.to_s
+    end
+
+    private
+
+    def wikidata_ids(links)
+      WikiData.ids_from_pages(language, links)
+    end
+
+    def language
+      'en'
```

Comment:
I don't think we can hardcode this to 'en' here. This will need to derive this from the `url` of the page being scraped. (A more advanced version will need to cope with interwiki links, but that can come later)